gengogettext.py

```diff
@@ -186,86 +188,149 @@ def update_db():
         order.save()
 
 
+def grouper(iterable, n, fillvalue=None):
+    "Collect data into fixed-length chunks or blocks"
+    # grouper('ABCDEFG', 3, 'x') --> ABC DEF Gxx
+    args = [iter(iterable)] * n
+    return itertools.izip_longest(fillvalue=fillvalue, *args)
+
+
 def update_statuses():
     print 'Updating state of in-progress jobs...'
-    jobs = {}
-    for job in Job.get_in_progress():
-        jobs[job.id] = job
-
-    r = gengo().getTranslationJobBatch(id=','.join(str(id) for id in jobs))
-    if r['response']:
-        for job_data in r['response']['jobs']:
-            job = jobs[int(job_data['job_id'])]
-            job.status = job_data['status']
-            job.source = job_data['body_src']
-            job.translation = job_data.get('body_tgt', '')
-            job.lang = job_data['lc_tgt']
-            if job.lang == 'no':
-                job.lang = 'nb'
-            job.save()
+    for batch in grouper(Job.get_in_progress(), 100):
+        jobs = {}
+        for job in batch:
+            if job:
+                jobs[job.id] = job
+
+        try:
+            r = gengo().getTranslationJobBatch(id=','.join(str(id) for id in jobs))
+        except ValueError:
+            print jobs.keys()
+            raise
+        if r['response']:
+            for job_data in r['response']['jobs']:
+                job = jobs[int(job_data['job_id'])]
+                job.status = job_data['status']
+                job.source = job_data['body_src']
+                job.translation = job_data.get('body_tgt', '')
+                job.lang = job_data['lc_tgt']
+                if job.lang == 'no':
+                    job.lang = 'nb'
+                job.save()
+
+
+def check_translation(job):
+    passed = True
+
+    if not job.translation.strip():
+        print "Empty translation"
+        passed = False
+
+    # Check that some strings, if present in the source, are present,
+    # identically, in the translation
+    for regex, message in (
+        (r'<.*?>', 'HTML tags'),
+        (r'%(?:\([a-zA-Z0-9_]+\))[#0 +-]?[0-9*]?\.?[0-9]?[diouxXeEfFgGcrs]',
```

Comment:
Sure, a class would be shorter :) Looking up exactly what's in classes takes longer than just writing the regex :P

I'm not particularly wedded to these regexes. Quite happy to borrow from elsewhere. But don't really want their named capture groups.